sdk/servicebus/service-bus/test/public/utils/testutils2.ts

```diff
@@ -290,13 +290,14 @@ export class ServiceBusTestHelpers {
   }
 
   async createTestEntities(
-    testClientType: TestClientType
+    testClientType: TestClientType,
+    requiresFresh: boolean = false
   ): Promise<ReturnType<typeof getEntityNames>> {
     // TODO: for now these aren't randomly named. This is prep so we can
     // do that soon.
     let entityValues = this._testClientEntities.get(testClientType);
 
-    if (entityValues == null) {
+    if (entityValues == null || requiresFresh) {
```

Comment:
![image](https://user-images.githubusercontent.com/10452642/110720381-4c154400-81c3-11eb-8640-5199e75424ac.png)

The test suite was failing because the service wasn't able to provide a session though the previous client was closed properly.
This new option allows overriding the caching as needed, which is false by default.